### DISCUSSION TITLE
Is there a way to programatically replay/trigger events?

### DISCUSSION THREAD

**Discussion by iampava** [microsoft/monaco-editor#3843] [Q&A] [answered]
The API has methods for listening to certain types of events. For example `onDidChangeModelContent` or `onDidChangeCursorSelection`. Is there a way to pass the payload of those events back to the Editor, and have it update accordingly?

To give you more context, let's say I save all event payloads of the `onDidChangeModelContent` and `onDidChangeCursorSelection` APIs in an Array. Is there a way for me to replay those events, basically iterating through the Array, calling a hypothetical `dispatchEvent(...)` method and then having the editor update?

**Comment by tamayika** [comment, marked as answer]
There is no method like that, but you can simulate it by `editor.executeEdits()` and `editor.setSelection()`;

**Reply by iampava** [reply]
Thanks!
